spec/javascripts/components/table-filter-spec.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-env jasmine, jquery */
+/* global GOVUK */
+
+describe('Table Filter component', function () {
+  'use strict'
+
+  var tableFilter
+  var searchInput
+  var element
+
+  beforeEach(function () {
+    var FIXTURE =
+      '<div data-module="table-filter">' +
+        '<div class="govuk-table-filter">' +
+          '<input name="filter">' +
+        '</div>' +
+        '<table>' +
+          '<thead></thead>' +
+          '<tbody class="govuk-table__body">' +
+            '<tr class="govuk-table__row">' +
+              '<td>bcd</td>' +
+              '<td>efg</td>' +
+              '<td>hio</td>' +
+            '</tr>' +
+            '<tr class="govuk-table__row">' +
+              '<td>klm</td>' +
+              '<td>nop</td>' +
+              '<td>£35</td>' +
+            '</tr>' +
+          '</tbody>' +
+        '</table>' +
+        '<p class="govuk-table__message govuk-table__message--hidden">That search returns no results.</p>' +
+      '</div>'
+
+    window.setFixtures(FIXTURE)
+
+    element = document.querySelector('[data-module="table-filter"]')
+    searchInput = document.querySelector('input')
+    tableFilter = new GOVUK.Modules.TableFilter(element)
+  })
+
+  it('Calls the `updateRows` method when the user types in the input field', function () {
```

Comment:
Why is it important to test that a specific function within the code has been called? Is it not enough to test that the expected outcome happens when certain things are put into the search box? (which is what the next test does anyway)

Reply:
We are testing here that the event listener has been added when the component is initialised. I have updated the description to better reflect that. The test that follows deals with the outcome of calling the method. By using separate tests for these two things we are breaking the dependency of one on the other. I’m happy to be convinced otherwise but for me that is a more valuable way to test the overall functionality: if the component as a whole breaks it is clearer where the problem has occurred.

Reply:
That makes sense.